docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/openebs-on-kubernetes-platforms/talos.md

```diff
@@ -10,45 +10,85 @@ description: This section explains about the Platform Support for Replicated PV
 ---
 # Replicated PV Mayastor Installation on Talos
 
-## Preparing the Nodes
-
-1. 2MiB-sized Huge Pages must be supported and enabled on the mayastor storage nodes. A minimum number of 1024 such pages (i.e. 2GiB total) must be available exclusively to the Mayastor pod on each node.
-
 :::info
-HugePages and node labels can be configured either during initial cluster creation or on running worker nodes.
+All the below configurations can be configured either during initial cluster creation or on running worker nodes.
 :::
 
-2. Set the `vm.nr_hugepages` sysctl and add `openebs.io/engine=mayastor` labels to the nodes which are meant to be storage nodes. This can be done with `talosctl edit/patch machineconfig` or via config patches during `talosctl gen config`.
+## Talos Control Plane Changes
 
-**Example**
+### Pod Security
 
-1. By using talosctl gen config:
+By default, Talos Linux applies a baseline pod security profile across namespaces except for the kube-system namespace. This default setting restricts Replicated PV Mayastors’s ability to manage and access system resources. You need to add the exemptions for Replicated PV Mayastor namespace. See the [Talos Documentation](https://www.talos.dev/v1.6/kubernetes-guides/configuration/pod-security/) for detailed instructions on Pod Security.
 
-    - Create a config patch file `mayastor-patch.yaml`.
+**Create a file cp.yaml**
 
-    ```
-    - op: add
-    path: /machine/sysctls
-    value:
-        vm.nr_hugepages: "1024"
-    - op: add
-    path: /machine/nodeLabels
-    value:
-        openebs.io/engine: mayastor
-    ```
+```
+- op: add
+  path: /cluster/apiServer/admissionControl
+  value:
+    - name: PodSecurity
+      configuration:
+        exemptions: null
+        namespaces:
+          - mayastor
+```
+
+## Talos Worker Node Changes
+
+### Huge Pages
+
+2MiB-sized Huge Pages must be supported and enabled on the Replicated PV Mayastor storage nodes. A minimum number of 1024 such pages (i.e. 2GiB total) must be available exclusively to the  Replicated PV Mayastor pod on each node.
+
+### Labels
+
+The OpenEBS storage type "Replicated PV Mayastor" must be labeled to all worker nodes that will have IO engine pods running on them. This label will be used as a node selector by the IO engine Daemonset that is deployed as a part of the Replicated PV Mayastor data plane components installation.
```

Comment:
The label is not "Replicated PV Mayastor". It is infact, 
_openebs.io/engine: "mayastor"_